wallet-new/src/Cardano/Wallet/Kernel/DB/AcidState.hs

```diff
@@ -105,26 +105,28 @@ newPending accountId tx = runUpdate' . zoom dbHdWallets $
     zoom hdAccountCheckpoints $
       mapUpdateErrors NewPendingFailed $ Spec.newPending tx
 
--- | Errors thrown by 'cancelPending'
-data CancelPendingError =
-    -- | Unknown account
-    CancelPendingUnknown UnknownHdAccount
-
-deriveSafeCopy 1 'base ''CancelPendingError
-
 -- | Cancels the input transactions from the 'Checkpoints' of each of
 -- the accounts cointained in the 'Cancelled' map.
 --
 -- The reason why this function doesn't take a 'HdRootId' as an argument
 -- is because the submission layer doesn't have the notion of \"which HdWallet
 -- is this transaction associated with?\", but it merely dispatch and cancels
 -- transactions for all the wallets managed by this edge node.
-cancelPending :: InDb Cancelled -> Update DB (Either CancelPendingError ())
-cancelPending (InDb cancelled) = runUpdate' . zoom dbHdWallets $
-    forM_ (Map.toList cancelled) $ \(accountId, txids) -> do
-        zoomHdAccountId CancelPendingUnknown accountId $ do
-          account <- get
-          put (over hdAccountCheckpoints (Spec.cancelPending txids) account)
+cancelPending :: Map HdAccountId (InDb (Set Core.TxId)) -> Update DB ()
+cancelPending cancelled = void . runUpdate' . zoom dbHdWallets $
+    forM_ (Map.toList cancelled) $ \(accountId, InDb txids) ->
+        -- Here we are deliberately swallowing the possible exception
+        -- returned by the wrapped 'zoom' as the only reason why this update
+        -- might fail is if, in the meantime, the target account was cancelled,
+        -- in which case we do want the entire computation to abort, but the
+        -- errored computation to simply be skipped.
+        handleError (\(_e :: UnknownHdAccount) -> return ()) $
+            zoomHdAccountId identity accountId $ do
+              account <- get
+              put (over hdAccountCheckpoints (Spec.cancelPending txids) account)
```

Comment:
Not a big fan of this `get .. put` idiom... why not `modify`? (I know, we're in a transaction, it's the same thing. but still.)

Reply:
I like modify as well -- I was merely playing with the implementation and when it type checked -- I left it like so. I laziness sin, if you wish. Will amend 😉